### DISCUSSION TITLE
Possible to get typed error from convenient error?

### DISCUSSION THREAD

**Discussion by mattoni** [ogen-go/ogen#1108] [Q&A]
Using the client, is it possible to get the API response out of a 'convenient error' or should this functionality be disabled if we want to do anything other than print a JSON error?

**Comment by ghost** [comment]
`Client` type returns convenient error response as `error`. You can get the response structure by using [`errors.As`](https://pkg.go.dev/errors#As).
```go
var apiError api.ErrorStatusCode

resp, err := client.Operation(ctx)
switch {
case errors.As(err, &apiError):
    // API error.
    fmt.Println(apiError.Message)
case err != nil:
    // Any other error.
default:
    // No error.
}
```
